content/articles/constructors-in-kotlin/index.md

```diff
@@ -0,0 +1,180 @@
+- [Types of constructors in Kotlin](#types-of-constructors-in-kotlin)
+- [Init blocks](#init-blocks)
+- [Best practices for object-oriented programming](#best-practices-for-object-oriented-programming)
+
+### Prerequisites
+To fully understand the article, you are required to have the following:
+
+- The Java Development Kit (JDK) installed on your computer
+- A basic understanding of programming languages
+- The IntelliJ code editor
+
+### What is a constructor?
+A constructor is a block of code that is similar to a procedure or a method, usually declared alongside the class. They are mostly used when you need to initialize variables while creating an object.
+
+### How to use constructors to create an object
+As a function, the primary constructor must be defined with the same name as the class, followed by parentheses. However, the secondary constructor must be expressed with the `constructor` keyword.
+
+### Types of constructors in Kotlin
+There are two different kinds of constructors in Kotlin, which are:
+
+- The primary constructor
+- The secondary constructor
+
+#### Primary constructors
+The primary constructor is a section of the class header that comes after the class name. The constructor keyword is used after the class name. The example below exemplifies the definition:
+
+```kotlin
+class student Constructor(name:string,age:Int,Email:string)
+ { 
+     ///code to be processed
+}
+```
+The constructor keyword can be ignored if the primary constructor has no annotations or accessibility modifiers:
+
+In the preceding example, we are not going to use the `constructor` keyword. By default, all the constructors are public which means they are visible everywhere in the class. Therefore we don't have to include the `constructor` keyword, as shown below:
+
+```kotlin
+class student(name:string,age:Int,Email:string){ }
+```
+
+#### Secondary constructor
+In Kotlin, a secondary constructor is created using the `constructor` keyword. In addition, the secondary constructor can be used to call the primary constructor. The example below shows how you can declare two constructors with a student having two parameters (name and year). 
+
+```kotlin
+class student{ 
+    constructor(name: Int){ 
+
+    } 
+    constructor(year:Int, id: Int){ 
+
+    } 
+}
+```
+
+Also, the secondary constructor can assign values to fields while the class is instantiated. The example below illustrates this property:
+
+```kotlin
+class myStudent{ 
+    constructor(name:String,id:Int{
+         println("Name=${name}") 
+         println("Id=${id}")
+    } 
+         
+    fun main(args: Array <string>) { 
+        val mystudent=myClass("Alex",200) 
+    }
+}
+```
+
+The output of the above code will look like this:
+
+```
+Name = Alex 
+Id = 200
+```
+
+Kotlin also offers room for using a primary constructor with a secondary constructor in the same class. You need to use the `this` keyword for authorization:

```

Comment:
> You need to use the `this` keyword for authorization

Not very descriptive of what is happening here. The `this` keyword is to call the primary constructor. You should mention this.